nusantara/nusa_datasets/sentiment_nathasa_review/sentiment_nathasa_review.py

```diff
@@ -0,0 +1,138 @@
+  journal={ITSMART: Jurnal Teknologi dan Informasi},
+  volume={6},
+  number={2},
+  pages={92--97},
+  year={2018}
+}
+"""
+
+_DATASETNAME = "sentiment_nathasa_review"
+
+_DESCRIPTION = """\
+Customer Review (Natasha Skincare) is a customers emotion dataset, with amounted to 19,253 samples with the division for each class is 804 joy, 43 surprise, 154 anger, 61 fear, 287 sad, 167 disgust, and 17736 no-emotions.
+"""
+
+_HOMEPAGE = "https://jurnal.uns.ac.id/itsmart/article/viewFile/17328/15082"
+
+_LICENSE = "Unknown"
+
+_URLS = {
+    _DATASETNAME: "https://drive.google.com/uc?id=1D1pHX7CxrI-eIl2bAvIp1bWQeucyUGw0",
+}
+
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]
+
+_SOURCE_VERSION = "1.0.0"
+
+_NUSANTARA_VERSION = "1.0.0"
+
+class SentimentNathasaReview(datasets.GeneratorBasedBuilder):
+    """Customer Review (Natasha Skincare) is a customers emotion dataset, with amounted to 19,253 samples with the division for each class is 804 joy, 43 surprise, 154 anger, 61 fear, 287 sad, 167 disgust, and 17736 no-emotions."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    NUSANTARA_VERSION = datasets.Version(_NUSANTARA_VERSION)
+
+    BUILDER_CONFIGS = [
+        NusantaraConfig(
+            name="sentiment_nathasa_review_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description="sentiment_nathasa_review source schema",
+            schema="source",
+            subset_id="sentiment_nathasa_review",
+        ),
+        NusantaraConfig(
+            name="sentiment_nathasa_review_nusantara_text",
+            version=datasets.Version(_NUSANTARA_VERSION),
+            description="sentiment_nathasa_review Nusantara schema",
+            schema="nusantara_text",
+            subset_id="sentiment_nathasa_review",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "sentiment_nathasa_review_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+               {
+                    "id": datasets.Value("string"),
+                    "usr": datasets.Value("string"),
+                    "source": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "label": datasets.Value("string"),
+               }
+            )
+        elif self.config.schema == "nusantara_text":
+            features = schemas.text_features(['NOEMOTION', 'SURPRISE', 'SAD', 'JOY', 'FEAR', 'DISGUST', 'ANGER'])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        data_dir = Path(dl_manager.download(_URLS[_DATASETNAME]))
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
```

Comment:
Is the dataset unsplitted?

Could you maybe check the unique values in the "source" dict values?
"DATA_LATIH" is actually a signifier that this row is a TRAIN split, and "DATA_UJI" I think belong to TEST split.

Reply:
The source file is a csv file without header. I will check more about this in the original paper.
Btw, do you mean "LATIH" means "train" and "DATA_UJI" means test? I can't understand them actually.

Reply:
Yes, and specifically, "DATA LATIH" is "train data" and "DATA UJI" means "test data", you're correct.
To clarify, the rows below are samples of train data and test data
- Sample of train data: `(2, 'yandri_a04', 'NOEMOTION', 'DATA LATIH', 'suntik vitamin brapa')`
- Sample of test data: `(292, 'ochiiawaw', 'SURPRISE', 'DATA UJI', 'natasha jakal jogja facial dsini mbak nangani kasar bgt gak halus pdhal ad lho gak kyak gni')`

It's already amazing for me that you're working on a language that you're not even familiar with it. Thanks a lot @TysonYu for that 😁
Unfortunately, not only that you don't understand the language of the data, the paper is actually also written in Indonesian, mostly except abstract. So maybe that also won't help wither sadly. I think it's simpler if we just split it by recognizing the "DATA LATIH" and "DATA UJI" words in the data.

Is this maybe ok @TysonYu ?

Reply:
Hi @TysonYu , just to confirm that the 4th column of the CSV file is the dataset split.
So, please split the dataset according to that column. if it is `DATA LATIH`, it should go to the training set, while for `DATA UJI`, it should go to the test set. 

I think the easiest way to implement it is to pass the same `filepath` to both split, and filter the row based on the 4th column. I think something like the following should be enough.
```
test_data = list(filter(lambda x: 'DATA UJI' in x, open(filepath).readlines()))
```

Reply:
Thank you both! @bryanwilie and @SamuelCahyawijaya 

I will split the data in this way.

Reply:
Done